src/Microsoft.AspNetCore.SystemWebAdapters/Authentication/IRemoteAppAuthenticationResultProcessor.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+/// <summary>
+/// An interface for types that can modify a remote authentication result after
+/// it has been retrieved from a remote service. This interface allows users
+/// to specify heuristics to apply to remote authentication results to clean them up
+/// or modify their properties in an extensible way.
+/// </summary>
+public interface IRemoteAppAuthenticationResultProcessor
```

Comment:
Do we expect people to need to implement this?

Reply:
This one, I think users may want to implement. If their ASP.NET app has any sort of custom authentication that would return different headers or status codes than expected, implementing this interface allows the user to tweak the response object accordingly.